### DISCUSSION TITLE
Websearch completion API endpoint returns a "400 Bad Request" error when attempting to generate search queries.

### DISCUSSION THREAD

**Discussion by chrisoutwright** [open-webui/open-webui#3246] [General]
# Bug Report
The completion API endpoint returns a "400 Bad Request" error when attempting to generate search queries.

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]

**Steps to Reproduce:**

1. Search Query via SearXNG setup (different server in my case, all works), then using the codestral:latest model
2. webui send a POST request to the /api/task/query/completions endpoint.
3. Observe the "400 Bad Request" error in the server logs.

**Expected Behavior:**
The API should process the request and return a completion for the search query.

**Actual Behavior:**
The API returns a "400 Bad Request" error, indicating that the request is malformed or invalid.

## Environment

- Open WebUI Version: v0.3.5
- Ollama (if applicable): 0.1.41
- Operating System: Windows 10

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots
```
INFO: 192.168.1.224:61335 - "POST /api/chat/completed HTTP/1.1" 200 OK
INFO: 192.168.1.224:61335 - "POST /api/v1/chats/ed344228-dfbb-4676-9e46-d0c1019fbca8 HTTP/1.1" 200 OK
INFO: 192.168.1.224:61335 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
generate_title
codestral
INFO
.ollama.main:url: http://192.168.1.40:11436/
INFO: 192.168.1.224:61335 - "POST /api/task/title/completions HTTP/1.1" 200 OK
INFO: 192.168.1.224:61335 - "POST /api/v1/chats/ed344228-dfbb-4676-9e46-d0c1019fbca8 HTTP/1.1" 200 OK
INFO: 192.168.1.224:61335 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO: 192.168.1.224:61335 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
['vkUrntTOf4wDb9PqAAAB']
INFO: 192.168.1.224:61338 - "GET /_app/immutable/nodes/4.6db5c43b.js HTTP/1.1" 304 Not Modified
INFO: 192.168.1.224:61339 - "GET /_app/immutable/nodes/9.e324168b.js HTTP/1.1" 304 Not Modified
INFO: 192.168.1.224:61339 - "GET /ollama/api/version HTTP/1.1" 200 OK
generate_search_query
INFO: 192.168.1.224:61343 - "POST /api/task/query/completions HTTP/1.1" 400 Bad Request
generate_search_query
INFO: 192.168.1.224:61344 - "POST /api/task/query/completions HTTP/1.1" 400 Bad Request
INFO:root
to web search with ('searxng', 'who won EM 2024 france austria')
INFO:root
to web search with ('searxng', 'who won EM 2024 france austria')

```
**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Manual installation

## Additional Information

The issue seems to occur specifically when generating search query completions. Other API endpoints, such as chat completions, appear to work fine. This suggests the issue might be specific to the /api/task/query/completions endpoint or related request handling.

**Comment by luguoyixiazi** [comment]
This is beacuse of 
```
@app.post("/api/task/query/completions")
async def generate_search_query(form_data: dict, user=Depends(get_verified_user)):
    if len(form_data["prompt"]) < app.state.config.SEARCH_QUERY_PROMPT_LENGTH_THRESHOLD:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"Skip search query generation for short prompts (< {app.state.config.SEARCH_QUERY_PROMPT_LENGTH_THRESHOLD} characters)",
        )
```
in app/backend/main.py, 
and app.state.config.SEARCH_QUERY_PROMPT_LENGTH_THRESHOLD is 100……so……you know……

**Reply by chrisoutwright** [reply]
I though it would take in some from the chat to enrich the Search generation! I wondered about that Bad Request and about the literal search without taking in the context.

**Reply by luguoyixiazi** [reply]
In fact, when I comment out those code, it will do websearch, so it may be a measure used to determine whether to proceed with websearch, as is is also difficult to determine this it without model‘s help

**Reply by chrisoutwright** [reply]
You mean that there was an option to consider the full context with prior messages for a websearch as well, or could one add such a parameter (like consider N last chat items?) It would probably make the search query more relevant I would say (exponential decay of context relevance assumed, but since transformer based, self-attention would be at work anyway.).

**Comment by brenzel** [comment]
I have solved the problem by adjusting the settings.yml from searxng.

Add json to search/formats:

search:
  safe_search: 0
  autocomplete: ""
  default_lang: ""
  formats:
    - html
    - json


... and change bind_address to 0.0.0.0 (I use linux) and port to 8080:

server:
  limiter: false
  image_proxy: true
  port: 8080
  bind_address: "0.0.0.0"

**Comment by aniqueta** [comment]
This issue seems to have re-emerged for me in v0.3.13. Specifically this error for Web Search:
```
"POST /api/task/query/completions HTTP/1.1" 400 Bad Request
```

Wonder if it has for anyone else too?

**Comment by foxworld306** [comment]
Same issue in 0.3.13

**Reply by aniqueta** [reply]
Seems to be resolved for me in 0.3.15

**Reply by foxworld306** [reply]
upgraded to 0.3.15, not resolved for me...

**Comment by zhanghao-njmu** [comment]
Same issue in 0.3.15

**Comment by foxworld306** [comment]
I lowered the value from 100 to 5 (it was sufficiently low), and this 400 error code does not appear again. However, the web search still isn't functioning optimally; it frequently stops responding. I've been encountering RAG problems starting from version 0.3.8, which have persisted for me.  Ref: https://github.com/open-webui/open-webui/discussions/4779
As a result, I've reverted to version 0.3.5, and for now, both the web search and RAG functionalities that are critical to me are running smoothly in this version. Therefore, I'm currently awaiting the release of version 0.4...
![image](https://github.com/user-attachments/assets/4b5fea0f-1977-484b-a0eb-75ef422fbc5a)
